[IMP] l10n_se: Merge of code from xclude and linserv, update to BAS2020

larmar opened the pull request:
Description of the issue/feature this PR addresses:
Update of l10n_se

Current behavior before PR:
Som incorect accounts added and missing taxes following BAS2020 for AB

Desired behavior after PR is merged:
More correct accounting



--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

jco-odoo commented:
@larmar If it is to improve and to make it better compatible with BAS2020, we are certainly open to it and we can still update quite some things, but there are some things to be taken into account: 

- Please do not change all the ids of the account templates, tax groups, ...
- The country group non-Europe is not necessary: this works normally the way it should in v13.  The priority of the way the fiscal positions are applied, takes care of that. 
- You can not just change the tag_names in the tax report in v13.  Is what you are having so much different than what is already there in terms of the numbers reported? 
- Do you have the Profit and Loss and Balance Sheet themselves also?  We don't want to add the tags if we don't have the corresponding reports at least.  

Maybe we can check together with @daniel-stenlov to get it right for all of us?

larmar commented:
- I have updated the CoA and the ids are not changed.

- Noted, removed non-Europe

- I think they are the same or at least simmilar, removed. I will test. If not the same it's incorrect :-)

- Yes we have all legal reporting, but that should not be part of l10n_se ? They are part of l10n_se_reports or ?
For now i removed the tags

daniel-stenlov commented on addons/l10n_se/__manifest__.py:
```diff
@@ -1,21 +1,41 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+# Copyright (C) 2018 Linserv Aktiebolag, Sweden (<https://www.linserv.se>).
+
```
Please remove copyright notice. You are not the original creator of this account table and can therefor not claim copyright on it.

larmar replied:
It could be removed. Even that the file is completly different.

daniel-stenlov commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -52,21 +52,17 @@
-"a1920","1920","PlusGiro","account.data_account_type_liquidity","l10nse_chart_template","False"
-"a1930","1930","Företagskonto/checkkonto/affärskonto","account.data_account_type_liquidity","l10nse_chart_template","True"
-"a1940","1940","Övriga bankkonton","account.data_account_type_liquidity","l10nse_chart_template","False"
-"a2070","2070","Ändamålsbestämda medel","account.data_account_type_equity","l10nse_chart_template","False"
```
1910, 1920, 1930, 1940 and 2070 is included as recommended minimum for BAS 2020.

larmar replied:
2070 is for "Eget kapital i ideella föreningar och stiftelser"
Then 2060 should also be included ?

Any comment @daniel-stenlov

daniel-stenlov replied:
Don't touch it, it will interfere with the roadmap.

larmar replied:
@daniel-stenlov 
The question is if 2070 is included (not used by ltd companies)
should we include 2060 as well.
Not sure what roadmap we are talking about ?

daniel-stenlov commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -125,7 +121,6 @@
 "a2890","2890","Övriga kortfristiga skulder","account.data_account_type_current_liabilities","l10nse_chart_template","False"
 "a2910","2910","Upplupna löner","account.data_account_type_current_liabilities","l10nse_chart_template","False"
 "a2920","2920","Upplupna semesterlöner","account.data_account_type_current_liabilities","l10nse_chart_template","False"
-"a2930","2930","Upplupna pensionskostnader","account.data_account_type_current_liabilities","l10nse_chart_template","False"
```
This is not listed as the recommended minimum. However, it's included in the complete BAS 2020 and should not be removed.

larmar replied:
Its not in minimum and should not be included.
There are may accounts in extended

larmar replied:
Not in minimum ( https://github.com/odoo/odoo/pull/53092#discussion_r454259845 )

daniel-stenlov commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -220,27 +214,24 @@
 "a5410","5410","Förbrukningsinventarier","account.data_account_type_expenses","l10nse_chart_template","False"
 "a5420","5420","Programvaror","account.data_account_type_expenses","l10nse_chart_template","False"
 "a5460","5460","Förbrukningsmaterial","account.data_account_type_expenses","l10nse_chart_template","False"
-"a5480","5480","Arbetskläder","account.data_account_type_expenses","l10nse_chart_template","False"
```
This is not listed as the recommended minimum. However, it's included in the complete BAS 2020 and should not be removed.

larmar replied:
There is no logic of adding just some of the accounts in the extended list.

larmar replied:
Not in minimum ( https://github.com/odoo/odoo/pull/53092#discussion_r454259845 )

daniel-stenlov commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -220,27 +214,24 @@
 "a6090","6090","Övriga försäljningskostnader ","account.data_account_type_expenses","l10nse_chart_template","False"
 "a6100","6100","Kontorsmateriel och trycksaker (gruppkonto)","account.data_account_type_expenses","l10nse_chart_template","False"
 "a6210","6210","Telekommunikation","account.data_account_type_expenses","l10nse_chart_template","False"
-"a6230","6230","Datakommunikation","account.data_account_type_expenses","l10nse_chart_template","False"
```
This is not listed as the recommended minimum. However, it's included in the complete BAS 2020 and should not be removed.

larmar replied:
Not in minimum ( https://github.com/odoo/odoo/pull/53092#discussion_r454259845 )

daniel-stenlov commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -252,7 +243,6 @@
 "a6550","6550","Konsultarvoden","account.data_account_type_expenses","l10nse_chart_template","False"
 "a6560","6560","Serviceavgifter till branschorganisationer","account.data_account_type_expenses","l10nse_chart_template","False"
 "a6570","6570","Bankkostnader","account.data_account_type_expenses","l10nse_chart_template","False"
-"a6580","6580","Advokat- och rättegångskostnader","account.data_account_type_expenses","l10nse_chart_template","False"
```
This is not listed as the recommended minimum. However, it's included in the complete BAS 2020 and should not be removed.

larmar replied:
Not in minimum ( https://github.com/odoo/odoo/pull/53092#discussion_r454259845 )

daniel-stenlov commented on addons/l10n_se/data/account_chart_template_post_data.xml:
```diff
@@ -1,17 +1,23 @@
-        </record> 
+            <field name="default_cash_difference_income_account_id" ref="a3740"/>
+            <field name="default_cash_difference_expense_account_id" ref="a3740"/>
+            <field name="default_pos_receivable_account_id" ref="a1510"/>
```
This is not the default POS receivable account. Normal POS bookkeeping is usually done like this:

For CASH
1910  |  1658,00 SEK
2610  |  -240,00 SEK
2620  |  -49,07 SEK
3001  | -960,00 SEK
3002  | -408,93 SEK

For Creditcard:
1580  |  1658,00 SEK
2610  |  -240,00 SEK
2620  |  -49,07 SEK
3001  | -960,00 SEK
3002  | -408,93 SEK

When payment is registerd at bankaccount:
1580  |  -1658,00 SEK
1930  |  1658,00 SEK

1510 is used for invoices not POS. So it should be 1910 or 1580, not 1510.

larmar replied:
The 1580 is used for credit card. Not for cash.

The 1910 may be correct also, but its not a receivable account  and  not selectable in pos.
I don't think its in line with how odoo POS works.

in case of cash upon closing you get 2 bookings
3001 Credit
1510 Debit
And
1510 Credit
1910 Debit (If its the cash account)

You can argue its wrong, but I think is correct. 1510 is just in and out. 
The Invoice is directly payed.
You could say that odoo is issuing a "kontantfaktura" 
it's actually the same as a recipt.

You are issuing an invoice and in the same moment getting paid.

So the result would be the same as your example above.
So 1510 is not a bad option. 1910 would be more confusing.

larmar replied:
@jco-odoo 1910 is not a receivable account, and not correct.

jco-odoo replied:
@daniel-stenlov I agree with @larmar on this one: the pos receivable account should be of type receivable and 1910 is of type cash.

daniel-stenlov replied:
Sure..

larmar commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -177,7 +172,6 @@
 "a4212","4212","Sålda varor negativ VMB 25 %","account.data_account_type_direct_costs","l10nse_chart_template","False"
 "a4400","4400","Momspliktiga inköp i Sverige","account.data_account_type_direct_costs","l10nse_chart_template","False"
 "a4415","4415","Inköpta varor i Sverige, omvänd skattskyldighet, 25 %","account.data_account_type_direct_costs","l10nse_chart_template","False"
-"a4425","4425","Inköp tjänster i Sverige, omvänd skattskyldighet, 25 %","account.data_account_type_direct_costs","l10nse_chart_template","False"
```
same as above.

larmar replied:
Not in minimum ( https://github.com/odoo/odoo/pull/53092#discussion_r454259845 )

larmar commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -272,15 +262,14 @@
 "a7324","7324","Skattepliktiga traktamenten, utlandet","account.data_account_type_expenses","l10nse_chart_template","False"
 "a7331","7331","Skattefria bilersättningar","account.data_account_type_expenses","l10nse_chart_template","False"
 "a7332","7332","Skattepliktiga bilersättningar","account.data_account_type_expenses","l10nse_chart_template","False"
-"a7333","7333","Ersättning för trängselskatt, skattefri","account.data_account_type_expenses","l10nse_chart_template","False"
```
It's not a good idea to include accounts outside minimum, or the commonly used accounts in sweden.

The list would be extensive. The complete list would include several hundreds of accounts most not used by normal companies.

There exist flavors for different business types, Limited. Sole trader Associations etc.

The most commonly used is AB (Limited mid sized companies).
We only have this kind of companies in odoo today.

I think is better to add accounts for other kind of business upon need, you are free to amend the extra ones still following BAS.

Most companies would have a fit o 95-99% only using the  "Minimum".
Probably most feels a need to add some extra :-). 
There are also some commonly used in several accounting software

BAS website is a swedish stakholder assostation (https://www.bas.se/english/)
Most buisinesses are following this. Bigger and global companies may do different.
All of the swedish accounting systems are represented in this association

In the Swedish version the commonly used is dot marked. ( refered to as Minimum above )

To add additional accounts it may be good for a single user, the association has marked i "commonly used"
To use another year release could be possibly be argued.

So Its a commonly agreed vesion that is used by most companies to add extra account don't seems wise to me.

We both have removed most of the accounts not used by Limited companies, I guess for a reason ?

odoo is auto creating the bank accounts ranging 1910-1949 (in BAS). 
We have noted its not good to include this, better to let odoo do the work

But its better odoo decides whats best.

Regarding the rest of the accounts i don't think it's good for anyone to amend the standard
I beileve BAS has already done so, i would not arue what is a good or bad choise  and we can all have different opinion.

larmar replied:
@jco-odoo any comment on this ?

jco-odoo replied:
Better to stay to the minimum accounts and it is true Odoo will create bank accounts automatically, so the accounts might create confusion.  (need to check from a user perspective)

larmar replied:
@jco-odoo when can we merge this in the main ? I think the issues has been discussed and are solved

larmar commented on addons/l10n_se/data/account.account.template.csv:
```diff
@@ -220,27 +214,24 @@
 "a5612","5612","Försäkring och skatt för personbilar ","account.data_account_type_expenses","l10nse_chart_template","False"
 "a5613","5613","Reparation och underhåll av personbilar","account.data_account_type_expenses","l10nse_chart_template","False"
 "a5615","5615","Leasing av personbilar","account.data_account_type_expenses","l10nse_chart_template","False"
-"a5616","5616","Trängselskatt","account.data_account_type_expenses","l10nse_chart_template","False"
```
Not in minimum ( https://github.com/odoo/odoo/pull/53092#discussion_r454259845 )

jco-odoo reviewed (commented):
It is good to keep the diff to a minimum to be clear what is changed exactly.

jco-odoo commented:
outdated as @JulienAlardot could check thoroughly.
